modules/mediaConsortiumBidAdapter.js

```diff
@@ -0,0 +1,190 @@
+  supportedMediaTypes: [BANNER],
+  isBidRequestValid(bid) {
+    return true
+  },
+  buildRequests(bidRequests, bidderRequest) {
+    const {
+      auctionId,
+      bids,
+      gdprConsent: {gdprApplies = false, consentString} = {},
+      ortb2: {device, site}
+    } = bidderRequest
+    const currentTimestamp = Date.now()
+    const optimizations = getOptimizationsFromLocalStorage()
+
+    const impressions = bids.reduce((acc, bidRequest) => {
+      const {bidId, adUnitCode, mediaTypes} = bidRequest
+      const optimization = optimizations[adUnitCode]
+
+      if (optimization) {
+        const {expiresAt, isEnabled} = optimization
+
+        if (expiresAt >= currentTimestamp && !isEnabled) {
+          return acc
+        }
+      }
+
+      return acc.concat({id: bidId, adUnitCode, mediaTypes})
+    }, [])
+
+    if (!impressions.length) {
+      return
+    }
+
+    const request = {
+      id: auctionId ?? generateUUID(),
+      impressions,
+      device,
+      site,
+      user: {
+        ids: {}
+      },
+      regulations: {
+        gdpr: {
+          applies: gdprApplies,
+          consentString
+        }
+      },
+      timeout: 3600
+    }
+
+    const fpId = getFpIdFromLocalStorage()
+
+    if (fpId) {
+      request.user.ids['1plusX'] = fpId
+    }
+
+    const syncData = {
+      gdpr: gdprApplies,
+      ad_unit_codes: impressions.map(({adUnitCode}) => adUnitCode).join(',')
+    }
+
+    if (consentString) {
+      syncData.gdpr_consent = consentString
+    }
+
+    return [
+      {
+        method: 'GET',
+        url: SYNC_ENDPOINT,
+        data: syncData
+      },
+      {
+        method: 'POST',
+        url: AUCTION_ENDPOINT,
+        data: request
+      }
+    ]
+  },
+  interpretResponse(serverResponse, params) {
+    if (!isValidResponse(serverResponse)) return []
+
+    const {body: {bids, optimizations}} = serverResponse
+
+    if (optimizations && isArray(optimizations)) {
+      const currentTimestamp = Date.now()
+
+      const optimizationsToStore = optimizations.reduce((acc, optimization) => {
+        const {adUnitCode, isEnabled, ttl} = optimization
+
+        return {
+          ...acc,
+          [adUnitCode]: {isEnabled, expiresAt: currentTimestamp + ttl}
+        }
+      }, getOptimizationsFromLocalStorage())
+
+      localStorage.setItem(OPTIMIZATIONS_STORAGE_KEY, JSON.stringify(optimizationsToStore))
+    }
+
+    return bids.map((bid) => {
+      const {
+        impressionId,
+        price: {cpm, currency},
+        dealId,
+        ad: {
+          creative: {id, mediaType, size: {width, height}, markup}
+        },
+        ttl = 360
+      } = bid
+
+      const markupWithMacroReplaced = replaceAuctionPrice(markup, cpm)
+
+      return {
+        requestId: impressionId,
+        cpm,
+        currency,
+        dealId,
+        ttl,
+        netRevenue: true,
+        creativeId: id,
+        mediaType,
+        width,
+        height,
+        ad: markupWithMacroReplaced,
```

Comment:
I recommend you return the price with no macro, publishers could fork and change what you pay them

Reply:
You mean replacing the `AUCTION_PRICE` macro server-side instead of doing it here ?

Reply:
Yes